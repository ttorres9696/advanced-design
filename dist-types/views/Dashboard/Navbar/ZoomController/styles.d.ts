declare const _default: (props?: any) => Record<"value" | "controller", string>;
export default _default;
