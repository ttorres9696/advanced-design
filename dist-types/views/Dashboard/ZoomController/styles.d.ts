declare const _default: (props?: any) => Record<"controller" | "zoomButton" | "zoomInButton" | "zoomOutButton", string>;
export default _default;
