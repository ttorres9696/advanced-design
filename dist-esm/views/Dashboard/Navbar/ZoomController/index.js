"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _core = require("@material-ui/core");

var _icons = require("@material-ui/icons");

var _decimal = _interopRequireDefault(require("decimal.js"));

var _react = _interopRequireWildcard(require("react"));

var _reactRedux = require("react-redux");

var _actions = require("../../../../redux/canvas/actions");

var _styles = _interopRequireDefault(require("./styles"));

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ZoomController = function ZoomController() {
  var classes = (0, _styles.default)();
  var dispatch = (0, _reactRedux.useDispatch)();
  var scale = (0, _reactRedux.useSelector)(function (state) {
    return state.canvas.scale;
  });
  var onZoomInClick = (0, _react.useCallback)(function () {
    return dispatch((0, _actions.increaseZoom)());
  }, [dispatch]);
  var onZoomOutClick = (0, _react.useCallback)(function () {
    return dispatch((0, _actions.decreaseZoom)());
  }, [dispatch]);
  return /*#__PURE__*/_react.default.createElement(_core.Box, {
    className: classes.controller
  }, /*#__PURE__*/_react.default.createElement(_core.IconButton, {
    color: "inherit",
    "aria-label": "decrease-zoom",
    onClick: onZoomOutClick
  }, /*#__PURE__*/_react.default.createElement(_icons.RemoveCircleOutline, null)), /*#__PURE__*/_react.default.createElement(_core.Box, {
    className: classes.value
  }, new _decimal.default(scale).mul(100).toFixed(0), "%"), /*#__PURE__*/_react.default.createElement(_core.IconButton, {
    color: "inherit",
    "aria-label": "increase-zoom",
    onClick: onZoomInClick
  }, /*#__PURE__*/_react.default.createElement(_icons.AddCircleOutline, null)));
};

var _default = /*#__PURE__*/_react.default.memo(ZoomController);

exports.default = _default;