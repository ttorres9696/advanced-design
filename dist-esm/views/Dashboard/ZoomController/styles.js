"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _styles = require("@material-ui/core/styles");

var _default = (0, _styles.makeStyles)(function (theme) {
  return (0, _styles.createStyles)({
    controller: {
      position: 'absolute',
      display: 'flex',
      flexDirection: 'column',
      alignItems: 'center',
      backgroundColor: '#D0D4DB',
      boxShadow: '2px 2px 4px #EEE',
      borderRadius: 2,
      color: '#000',
      border: '1px solid #D0D4DB',
      bottom: 20,
      right: 20,
      zIndex: 9999,
      '&:focus': {
        outline: 'none !important'
      }
    },
    zoomButton: {
      backgroundColor: '#FFF',
      padding: 10,
      border: 'none',
      '&:hover': {
        cursor: 'pointer'
      },
      '&:focus': {
        outline: 'none !important'
      },
      '&:disabled': {
        cursor: 'not-allowed'
      },
      '&:active': {
        backgroundColor: '#0566AD',
        color: '#FFF'
      }
    },
    zoomInButton: {
      marginBottom: 1,
      borderTopLeftRadius: 2,
      borderTopRightRadius: 2
    },
    zoomOutButton: {
      borderBottomLeftRadius: 2,
      borderBottomRightRadius: 2
    }
  });
});

exports.default = _default;