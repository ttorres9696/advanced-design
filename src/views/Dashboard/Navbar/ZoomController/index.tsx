import { Box, IconButton } from '@material-ui/core';
import { AddCircleOutline as IncreaseZoomIcon, RemoveCircleOutline as DecreaseZoomIcon } from '@material-ui/icons';
import Decimal from 'decimal.js';
import React, { useCallback } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import { RootState } from '../../../../redux';
import { decreaseZoom, increaseZoom } from '../../../../redux/canvas/actions';
import useStyles from './styles';

const ZoomController: React.FC = () => {
  const classes = useStyles();
  const dispatch = useDispatch();

  const scale = useSelector((state: RootState) => state.canvas.scale);

  const onZoomInClick = useCallback(() => dispatch(increaseZoom()), [dispatch]);
  const onZoomOutClick = useCallback(() => dispatch(decreaseZoom()), [dispatch]);

  return (
    <Box className={classes.controller}>
      <IconButton color="inherit" aria-label="decrease-zoom" onClick={onZoomOutClick}>
        <DecreaseZoomIcon />
      </IconButton>
      <Box className={classes.value}>{new Decimal(scale).mul(100).toFixed(0)}%</Box>
      <IconButton color="inherit" aria-label="increase-zoom" onClick={onZoomInClick}>
        <IncreaseZoomIcon />
      </IconButton>
    </Box>
  );
};

export default React.memo(ZoomController);
