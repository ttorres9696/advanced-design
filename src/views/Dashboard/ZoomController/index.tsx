import { Box } from '@material-ui/core';
import { Add as IncreaseZoomIcon, Remove as DecreaseZoomIcon } from '@material-ui/icons';
import React, { useCallback } from 'react';
import { useDispatch } from 'react-redux';

import { decreaseZoom, increaseZoom } from '../../../redux/canvas/actions';
import useStyles from './styles';

const ZoomController: React.FC = () => {
  const classes = useStyles();
  const dispatch = useDispatch();

  const onZoomInClick = useCallback(() => dispatch(increaseZoom()), [dispatch]);
  const onZoomOutClick = useCallback(() => dispatch(decreaseZoom()), [dispatch]);

  return (
    <Box className={classes.controller}>
      <button className={[classes.zoomButton, classes.zoomInButton].join(' ')} onClick={onZoomInClick}>
        <IncreaseZoomIcon />
      </button>
      <button className={[classes.zoomButton, classes.zoomOutButton].join(' ')} onClick={onZoomOutClick}>
        <DecreaseZoomIcon />
      </button>
    </Box>
  );
};

export default React.memo(ZoomController);
