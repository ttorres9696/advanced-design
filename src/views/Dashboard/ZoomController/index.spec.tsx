import { mount, ReactWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { Provider } from 'react-redux';

import mockStore from '../../../redux/specs/mocks/state.mock';
import ZoomController from '.';

let wrapper: ReactWrapper;

beforeEach(() => {
  wrapper = mount(
    <Provider store={mockStore}>
      <ZoomController />
    </Provider>,
    {},
  );
});

describe('ZoomController', () => {
  it('renders correctly', () => expect(toJson(wrapper)).toMatchSnapshot());
});
