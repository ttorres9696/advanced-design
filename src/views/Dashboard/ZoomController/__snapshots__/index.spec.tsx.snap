// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZoomController renders correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Memo(ZoomController)>
    <Styled(MuiBox)
      className="makeStyles-controller-1"
    >
      <div
        className="MuiBox-root MuiBox-root-4 makeStyles-controller-1"
      >
        <WithStyles(ForwardRef(IconButton))
          aria-label="decrease-zoom"
          color="inherit"
          onClick={[Function]}
        >
          <ForwardRef(IconButton)
            aria-label="decrease-zoom"
            classes={
              Object {
                "colorInherit": "MuiIconButton-colorInherit",
                "colorPrimary": "MuiIconButton-colorPrimary",
                "colorSecondary": "MuiIconButton-colorSecondary",
                "disabled": "Mui-disabled",
                "edgeEnd": "MuiIconButton-edgeEnd",
                "edgeStart": "MuiIconButton-edgeStart",
                "label": "MuiIconButton-label",
                "root": "MuiIconButton-root",
                "sizeSmall": "MuiIconButton-sizeSmall",
              }
            }
            color="inherit"
            onClick={[Function]}
          >
            <WithStyles(ForwardRef(ButtonBase))
              aria-label="decrease-zoom"
              centerRipple={true}
              className="MuiIconButton-root MuiIconButton-colorInherit"
              disabled={false}
              focusRipple={true}
              onClick={[Function]}
            >
              <ForwardRef(ButtonBase)
                aria-label="decrease-zoom"
                centerRipple={true}
                className="MuiIconButton-root MuiIconButton-colorInherit"
                classes={
                  Object {
                    "disabled": "Mui-disabled",
                    "focusVisible": "Mui-focusVisible",
                    "root": "MuiButtonBase-root",
                  }
                }
                disabled={false}
                focusRipple={true}
                onClick={[Function]}
              >
                <button
                  aria-label="decrease-zoom"
                  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <ForwardRef>
                      <WithStyles(ForwardRef(SvgIcon))>
                        <ForwardRef(SvgIcon)
                          classes={
                            Object {
                              "colorAction": "MuiSvgIcon-colorAction",
                              "colorDisabled": "MuiSvgIcon-colorDisabled",
                              "colorError": "MuiSvgIcon-colorError",
                              "colorPrimary": "MuiSvgIcon-colorPrimary",
                              "colorSecondary": "MuiSvgIcon-colorSecondary",
                              "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                              "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                              "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                              "root": "MuiSvgIcon-root",
                            }
                          }
                        >
                          <svg
                            aria-hidden="true"
                            className="MuiSvgIcon-root"
                            focusable="false"
                            role="presentation"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M7 11v2h10v-2H7zm5-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                            />
                          </svg>
                        </ForwardRef(SvgIcon)>
                      </WithStyles(ForwardRef(SvgIcon))>
                    </ForwardRef>
                  </span>
                  <NoSsr>
                    <WithStyles(undefined)
                      center={true}
                    >
                      <ForwardRef(TouchRipple)
                        center={true}
                        classes={
                          Object {
                            "child": "MuiTouchRipple-child",
                            "childLeaving": "MuiTouchRipple-childLeaving",
                            "childPulsate": "MuiTouchRipple-childPulsate",
                            "ripple": "MuiTouchRipple-ripple",
                            "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                            "rippleVisible": "MuiTouchRipple-rippleVisible",
                            "root": "MuiTouchRipple-root",
                          }
                        }
                      >
                        <span
                          className="MuiTouchRipple-root"
                        >
                          <TransitionGroup
                            childFactory={[Function]}
                            component={null}
                            exit={true}
                          />
                        </span>
                      </ForwardRef(TouchRipple)>
                    </WithStyles(undefined)>
                  </NoSsr>
                </button>
              </ForwardRef(ButtonBase)>
            </WithStyles(ForwardRef(ButtonBase))>
          </ForwardRef(IconButton)>
        </WithStyles(ForwardRef(IconButton))>
        <Styled(MuiBox)
          className="makeStyles-value-2"
        >
          <div
            className="MuiBox-root MuiBox-root-26 makeStyles-value-2"
          >
            100
            %
          </div>
        </Styled(MuiBox)>
        <WithStyles(ForwardRef(IconButton))
          aria-label="increase-zoom"
          color="inherit"
          onClick={[Function]}
        >
          <ForwardRef(IconButton)
            aria-label="increase-zoom"
            classes={
              Object {
                "colorInherit": "MuiIconButton-colorInherit",
                "colorPrimary": "MuiIconButton-colorPrimary",
                "colorSecondary": "MuiIconButton-colorSecondary",
                "disabled": "Mui-disabled",
                "edgeEnd": "MuiIconButton-edgeEnd",
                "edgeStart": "MuiIconButton-edgeStart",
                "label": "MuiIconButton-label",
                "root": "MuiIconButton-root",
                "sizeSmall": "MuiIconButton-sizeSmall",
              }
            }
            color="inherit"
            onClick={[Function]}
          >
            <WithStyles(ForwardRef(ButtonBase))
              aria-label="increase-zoom"
              centerRipple={true}
              className="MuiIconButton-root MuiIconButton-colorInherit"
              disabled={false}
              focusRipple={true}
              onClick={[Function]}
            >
              <ForwardRef(ButtonBase)
                aria-label="increase-zoom"
                centerRipple={true}
                className="MuiIconButton-root MuiIconButton-colorInherit"
                classes={
                  Object {
                    "disabled": "Mui-disabled",
                    "focusVisible": "Mui-focusVisible",
                    "root": "MuiButtonBase-root",
                  }
                }
                disabled={false}
                focusRipple={true}
                onClick={[Function]}
              >
                <button
                  aria-label="increase-zoom"
                  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <ForwardRef>
                      <WithStyles(ForwardRef(SvgIcon))>
                        <ForwardRef(SvgIcon)
                          classes={
                            Object {
                              "colorAction": "MuiSvgIcon-colorAction",
                              "colorDisabled": "MuiSvgIcon-colorDisabled",
                              "colorError": "MuiSvgIcon-colorError",
                              "colorPrimary": "MuiSvgIcon-colorPrimary",
                              "colorSecondary": "MuiSvgIcon-colorSecondary",
                              "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                              "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                              "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                              "root": "MuiSvgIcon-root",
                            }
                          }
                        >
                          <svg
                            aria-hidden="true"
                            className="MuiSvgIcon-root"
                            focusable="false"
                            role="presentation"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                            />
                          </svg>
                        </ForwardRef(SvgIcon)>
                      </WithStyles(ForwardRef(SvgIcon))>
                    </ForwardRef>
                  </span>
                  <NoSsr>
                    <WithStyles(undefined)
                      center={true}
                    >
                      <ForwardRef(TouchRipple)
                        center={true}
                        classes={
                          Object {
                            "child": "MuiTouchRipple-child",
                            "childLeaving": "MuiTouchRipple-childLeaving",
                            "childPulsate": "MuiTouchRipple-childPulsate",
                            "ripple": "MuiTouchRipple-ripple",
                            "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                            "rippleVisible": "MuiTouchRipple-rippleVisible",
                            "root": "MuiTouchRipple-root",
                          }
                        }
                      >
                        <span
                          className="MuiTouchRipple-root"
                        >
                          <TransitionGroup
                            childFactory={[Function]}
                            component={null}
                            exit={true}
                          />
                        </span>
                      </ForwardRef(TouchRipple)>
                    </WithStyles(undefined)>
                  </NoSsr>
                </button>
              </ForwardRef(ButtonBase)>
            </WithStyles(ForwardRef(ButtonBase))>
          </ForwardRef(IconButton)>
        </WithStyles(ForwardRef(IconButton))>
      </div>
    </Styled(MuiBox)>
  </Memo(ZoomController)>
</Provider>
`;
